stellargraph/data/explorer.py

```diff
@@ -47,6 +48,62 @@ def _default_if_none(value, default, name, ensure_not_none=True):
     return value
 
 
+def _ensure_int(value, name, min_value):
+    if not isinstance(value, int) or value < min_value:
+        raise ValueError(
+            f"{name}: expected an integer greater than or equal to {min_value}, found: {value}"
+        )
+
+
+class RandomWalk(ABC):
+    """
+    Abstract base class for Random Walk classes. A Random Walk class must implement a ``run`` method
+    which takes an iterable of node IDs and returns a list of walks. Each walk is a list of node IDs
+    that contains the starting node as its first element.
+    """
+
+    def __init__(self, graph, seed=None):
+        if not isinstance(graph, StellarGraph):
+            raise TypeError("Graph must be a StellarGraph or StellarDiGraph.")
+
+        self.graph = graph
+        self._random_state, self._np_random_state = random_state(seed)
+
+    def _get_random_state(self, seed):
+        """
+        Args:
+            seed: The optional seed value for a given run.
+
+        Returns:
+            The random state as determined by the seed.
+        """
+        if seed is None:
+            # Restore the random state
+            return self._random_state
+        # seed the random number generator
+        _ensure_int(seed, "seed", min_value=0)
+        rs, _ = random_state(seed)
+        return rs
+
+    @staticmethod
+    def _validate_walk_params(nodes, n, length):
+        if not is_real_iterable(nodes):
+            raise ValueError(f"nodes: expected an iterable, found: {nodes}")
+        if len(nodes) == 0:
+            warnings.warn(
+                "No root node IDs given. An empty list will be returned as a result.",
+                RuntimeWarning,
+                stacklevel=3,
+            )
+
+        _ensure_int(n, "n", min_value=1)
+        _ensure_int(length, "length", min_value=1)
+
+    @abstractmethod
+    def run(self, nodes, **kwargs):
```

Comment:
If we're (slightly) refactoring here, would

```suggestion
    def run(self, start_nodes, **kwargs):
```
be slightly better as an API? (not that it matters much at all)

Reply:
This would be a breaking change, since there are a number of notebooks that use `BiasedRandomWalk.run` directly for instance. Do you think it's worth doing?

Reply:
I think it's only breaking if they do `.run(nodes=...)`? If so, then let's leave it.

Reply:
I went through our `node2vec` and `metapath2vec` demos and they all seem to pass in `nodes` as a keyword argument when they run the random walk, so I'll leave it.